src/dict.c

```diff
@@ -675,23 +694,36 @@ void dictRelease(dict *d)
 
 dictEntry *dictFind(dict *d, const void *key)
 {
-    dictEntry *he;
-    uint64_t h, idx, table;
+    dictEntry *he = NULL;
+    uint64_t h, idx;
 
     if (dictSize(d) == 0) return NULL; /* dict is empty */
-    if (dictIsRehashing(d)) _dictRehashStep(d);
+
     h = dictHashKey(d, key);
-    for (table = 0; table <= 1; table++) {
+    idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[0]);
+    if ((long)idx >= d->rehashidx) {
+        he = d->ht_table[0][idx];
+    }
+
+    if (dictIsRehashing(d)) {
+        if ((long)idx >= d->rehashidx && he) {
+            dictBucketRehash(d, h);
```

Comment:
Why not passing `he` instead of `h`?

Reply:
We can also pass just `idx` not that I look at it
So it will be dict bucket rehash at an index in HT 0